dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/ElevateUserTrnVerification.cshtml.cs

```diff
@@ -0,0 +1,79 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeacherIdentity.AuthServer.Infrastructure.Security;
+using TeacherIdentity.AuthServer.Models;
+
+namespace TeacherIdentity.AuthServer.Pages.Admin;
+
+[Authorize(AuthorizationPolicies.GetAnIdentityAdmin)]
+public class ElevateUserTrnVerificationModel : PageModel
+{
+    private readonly TeacherIdentityServerDbContext _dbContext;
+    private readonly IClock _clock;
+    public new User? User { get; set; }
+    [FromRoute]
+    public Guid UserId { get; set; }
+
+    public ElevateUserTrnVerificationModel(TeacherIdentityServerDbContext dbContext, IClock clock)
+    {
+        _dbContext = dbContext;
+        _clock = clock;
+    }
+
+    public void OnGet()
+    {
+    }
+
+    public async Task<IActionResult> OnPost()
+    {
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        User!.TrnVerificationLevel = TrnVerificationLevel.Medium;
```

Comment:
This should assign `TrnAssociationSource` (to `SupportUi`) instead of `TrnVerificationLevel`

Reply:
Actually, we should probably set `TrnVerificationLevel` to `null` so it lines up with how users that didn't match at all look after a TRN is assigned.